### PR TITLE
fix: Only skip invalid meshes, instead of aborting the pass completely.

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/543)

### CODE REVIEW

File: amethyst_renderer/src/pass/pbm/separate.rs [hunk 1]
```diff
@@ -188,7 +188,7 @@ impl Pass for DrawPbmSeparate {
             {
                 match mesh.buffer(attrs) {
                     Some(vbuf) => effect.data.vertex_bufs.push(vbuf.clone()),
-                    None => return,
+                    None => continue 'drawable,
```
Comment (omni-viral) [inline amethyst_renderer/src/pass/pbm/separate.rs]: `effect.data.vertex_bufs`  must be cleared.

**Comment by Rhuagh** [comment]
Fixed.

**Comment by omni-viral** [comment]
@torkleyy I wouldn't spend much time to fix minor problems in old renderer.

**Comment by Rhuagh** [comment]
Yeah, there's some code duplication, but no good way of removing that without introducing about the same amount of duplication just trying to send data to functions.

**Comment by Xaeroxe** [comment]
Overriding

**Pull request merged**